PowerEditor/src/WinControls/StaticDialog/StaticDialog.cpp

```diff
@@ -79,10 +79,15 @@ void StaticDialog::goToCenter(UINT swpFlags)
 {
 	RECT rc{};
 	::GetClientRect(_hParent, &rc);
+	if ((rc.left == rc.right) || (rc.top == rc.bottom))
+		swpFlags |= SWP_NOSIZE; // sizing has no sense here
+
 	POINT center{};
 	center.x = rc.left + (rc.right - rc.left)/2;
 	center.y = rc.top + (rc.bottom - rc.top)/2;
 	::ClientToScreen(_hParent, &center);
+	if ((center.x == -32000) && (center.y == -32000))
+		swpFlags |= SWP_NOMOVE; // moving has no sense here (owner wnd is minimized)

```

Comment:
I restored `void StaticDialog::goToCenter(UINT swpFlags)` as it is in master and keep your modification in `int Notepad_plus::doSaveOrNot(const TCHAR* fn, bool isMulti)` - it fixes issue #14718.

So my questions here are: 
1. if the modification of `void StaticDialog::goToCenter(UINT swpFlags)` is necessary, what's the reason? 
2. Why -32000?

Reply:
> 1\. if the modification of void StaticDialog::goToCenter(UINT swpFlags) is necessary, what's the reason?

The `StaticDialog::goToCenter` fix is necessary to correct the very essence of the problem and not just one specific occurrence of it. You can try the opposite (restore the `Notepad_plus::doSaveOrNot` code and keep the fix for the `StaticDialog::goToCenter`) and the current problem will be solved too, but in this case also for any future possible occurrence of this situation (a N++ modal child/pop-up wnd centering while its owner-wnd is minimized).

The additional `Notepad_plus::doSaveOrNot` fix is only there to mimic the behavior when the N++ is minimized to systray instead to taskbar and the `doSaveOrNotBox` dlg is being invoked.

> 2\. Why -32000?

Windows NT family just stuck with that `-32000,-32000` coordinates for the minimized wnds, probably for some prehistoric compatibility reasons. I am copying here my explanation from the issue:
"it is also centered relative to the owner (main N++ wnd), but when that owner is minimized (systray seems to be ok), the `ClientToScreen` WINAPI returns [-32000, -32000](https://devblogs.microsoft.com/oldnewthing/20041028-00/?p=37453) as the `center` `x,y`"

Reply:
> The StaticDialog::goToCenter fix is necessary to correct the very essence of the problem and not just one specific occurrence of it. You can try the opposite (restore the Notepad_plus::doSaveOrNot code and keep the fix for the StaticDialog::goToCenter) and the current problem will be solved too, but in this case also for any future possible occurrence of this situation (a N++ modal child/pop-up wnd centering while its owner-wnd is minimized).

***"A good programmer is someone who always looks both ways before crossing a one-way street."***
:)

> "it is also centered relative to the owner (main N++ wnd), but when that owner is minimized (systray seems to be ok), the ClientToScreen WINAPI returns [-32000, -32000](https://devblogs.microsoft.com/oldnewthing/20041028-00/?p=37453) as the center x,y"

Then it would be great to add https://devblogs.microsoft.com/oldnewthing/20041028-00/?p=37453 in a comment to justify the magic number, wouldn't be it?

Reply:
@donho 
Done.